homeassistant/components/playstation_network/config_flow.py

```diff
@@ -132,3 +155,51 @@ async def async_step_reauth_confirm(
                 "psn_link": PSN_LINK,
             },
         )
+
+
+class FriendSubentryFlowHandler(ConfigSubentryFlow):
+    """Handle subentry flow for adding a friend."""
+
+    friends_list: dict[str, User]
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> SubentryFlowResult:
+        """Subentry user flow."""
+        errors: dict[str, str] = {}
+        config_entry: PlaystationNetworkConfigEntry = self._get_entry()
+        if user_input is None:
+            self.friends_list = await self.hass.async_add_executor_job(
+                lambda: {
+                    friend.account_id: friend
+                    for friend in config_entry.runtime_data.user_data.psn.user.friends_list()
+                }
+            )
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.friends_list[user_input[CONF_ACCOUNT_ID]].online_id,
+                data=user_input,
+                unique_id=user_input[CONF_ACCOUNT_ID],
+            )
```

Comment:
why do we have both statements? If `self.friend_list` is filled, why don't we have the `if user_input is not None` above the `user_input is None` (or maybe remove that if statement altogether)